src/Forms/ConfirmedPasswordField.php

```diff
@@ -562,9 +606,18 @@ public function saveInto(DataObjectInterface $record)
             return;
         }
 
-        if (!($this->canBeEmpty && !$this->value)) {
-            parent::saveInto($record);
+        // Create a random password for Member if password is blank
+        if (!$this->value
+            && $this->canBeEmpty
+            && $this->generateRandomPasswordOnEmpty
+        ) {
+            if (!is_callable($this->randomPasswordCallback)) {
+                throw new LogicException('randomPasswordCallback must be callable');
+            }
+            $this->value = call_user_func($this->randomPasswordCallback, ['maxLength' => $this->maxLength]);
```

Comment:
Having both a settable boolean _and_ a callback seems like a lot of complexity for something that realistically isn't likely to be used outside of our identified usecase for creating new `Member` records.

Reply:
The alternative is to remove the boolean and only have a $randomPasswordCallback and use that like a boolean in the conditional

I initially considered that though what made be go with the extra boolean was to strop it reading badly i.e. you go $field->setRandomPasswordCallback(<function>) to enable the random password functionality.  You kind of need a convoluted method name at that point like $field->setRandomPasswordCallbackToBeUsedWhenPasswordIsEmpty() so that it reads better.  I prefer just the extra boolean

Reply:
Hmm. I prefer not having the boolean and relying on PHPDoc to tell developers that having the callback  means the callback will be used, and not having the callback turns the funcitonality off. That means less API surface, and it's not something people are likely to use anyway. Plus having the boolean as its own method implies that it can be operated on its own, i.e. it implies that no callback is required.

I'll accept having both though if you add clear PHPDoc comments to all the methods explaining how they work.
In fact that probably should be done anyway - please add clear PHPDoc comments to all new public API.

Reply:
Have removed bool and added docblock to relevant API - though not to getters/setters and it better to add the docblock to the private property that's getting updated.

Side note ideally would just get rid of getters/setters and just use public properties though that's inconsistent with the rest of the code base so not doing here

Reply:
PHPDoc comments for public methods are surfaced in IDEs as well as in the api docs in api.silverstripe.org. Neither of those know to show the PHPDoc that you've added to the private properties. The PHPDoc for the private property might be useful for maintainers, but it's functionally useless for anyone who wants to just call the public methods.

Please add PHPDoc comments to the public methods.

Reply:
This still has not been done.

Reply:
Updated